From d20c0086f4d9ad97aa17b7220c64906306dc495a Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 7 Jun 2018 00:10:12 +0800
Subject: [PATCH 3936/5242] MLK-18535-10 drm/imx: lcdif: limit the maximum
 resolution to '1920x1080'

commit  dafae72dcf2617373174d2fb5f4880a9d357b152 from
https://source.codeaurora.org/external/imx/linux-imx.git

For now, the higher resolution than '1920x1080' is not supported
yet. So limit the 'max_width' and 'max_height' of mode_config
to be 1920 and 1080.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-crtc.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
index 0d76d90..9895544 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
@@ -295,6 +295,10 @@ static int lcdif_crtc_bind(struct device *dev, struct device *master,
 	if (!drm->mode_config.helper_private)
 		drm->mode_config.helper_private = &lcdif_drm_mode_config_helpers;
 
+	/* limit the max width and height */
+	drm->mode_config.max_width  = 1920;
+	drm->mode_config.max_height = 1080;
+
 	dev_set_drvdata(dev, lcdif_crtc);
 
 	dev_dbg(dev, "%s: lcdif crtc bind end\n", __func__);
-- 
1.7.9.5

